### PR TITLE
Adding Vespa examples

### PR DESCRIPTION
**Pull request by davidlago**
Adding a Vespa sample script and modified README with instructions. Example command line output after running what's in the README:

```sh
$> docker info | grep "Total Memory" # make sure it's at least 4Gb
docker run --detach --name vespa --hostname vespa-container \
  --publish 8080:8080 --publish 19071:19071 \
  vespaengine/vespa
 Total Memory: 7.667GiB
b381108b819a322dc36648cdbb9a9787c05a1b90557536e27f23526d475cd37b

$> (cd src/vespa/vector-app && zip -r - .) | \
  curl --header Content-Type:application/zip --data-binary @- \
  localhost:19071/application/v2/tenant/default/prepareandactivate
curl --header Content-Type:application/zip -XPOST localhost:19071/application/v2/tenant/default/session
  adding: services.xml (deflated 59%)
  adding: schemas/ (stored 0%)
  adding: schemas/vector.sd (deflated 62%)
{"log":[],"tenant":"default","session-id":"2","url":"http://localhost:19071/application/v2/tenant/default/application/default/environment/prod/region/default/instance/default","message":"Session 2 for tenant 'default' prepared and activated.","configChangeActions":{"restart":[],"refeed":[],"reindex":[]}}{"log":[],"tenant":"default","session-id":"3","prepared":"http://localhost:19071/application/v2/tenant/default/session/3/prepared","content":"http://localhost:19071/application/v2/tenant/default/session/3/content/","message":"Session 3 for tenant 'default' created."}%
$> ENDPOINT=http://localhost:8080 poetry run src/vespa/hello.py
> POST http://localhost:8080/document/v1/vector/vector/docid/vec1
< POST http://localhost:8080/document/v1/vector/vector/docid/vec1 - 200
> POST http://localhost:8080/document/v1/vector/vector/docid/vec2
< POST http://localhost:8080/document/v1/vector/vector/docid/vec2 - 200
> GET http://localhost:8080/search/?yql=select%20%2A%20from%20sources%20%2A%20where%20%7BtargetHits%3A%201%7DnearestNeighbor%28values%2Cvector_query_embedding%29&ranking.profile=vector_similarity&hits=1&input.query%28vector_query_embedding%29=%5B0.1%2C0.2%2C0.3%5D
< GET http://localhost:8080/search/?yql=select%20%2A%20from%20sources%20%2A%20where%20%7BtargetHits%3A%201%7DnearestNeighbor%28values%2Cvector_query_embedding%29&ranking.profile=vector_similarity&hits=1&input.query%28vector_query_embedding%29=%5B0.1%2C0.2%2C0.3%5D - 200
{'sddocname': 'vector', 'documentid': 'id:vector:vector::vec1', 'id': 'vec1', 'values': {'type': 'tensor<float>(x[3])', 'values': [0.10000000149011612, 0.20000000298023224, 0.30000001192092896]}, 'metadata': {'genre': 'drama'}}
> DELETE http://localhost:8080/document/v1/vector/vector/docid/vec1
< DELETE http://localhost:8080/document/v1/vector/vector/docid/vec1 - 200
> DELETE http://localhost:8080/document/v1/vector/vector/docid/vec2
< DELETE http://localhost:8080/document/v1/vector/vector/docid/vec2 - 200
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,6 +54,31 @@ USERNAME=admin PASSWORD=admin ENDPOINT=https://localhost:9200 poetry run src/ope
 < DELETE https://localhost:9200/my-index - 200
 ```
 
+### Vespa
```
Comment (dblock) [inline README.md:58]: Update the TOC at the top of the README.

File: README.md [hunk 2]
```diff
@@ -54,6 +54,31 @@ USERNAME=admin PASSWORD=admin ENDPOINT=https://localhost:9200 poetry run src/ope
...
+
+You can run Vespa in a [managed cloud service](https://cloud.vespa.ai/), but for this example we'll use again a Docker container:
```
Comment (dblock) [inline README.md]: Link the first occurrence of Vespa? 

Nit: I like to be succinct. Would say "Start Vespa in a Docker container. You may also run it in the vespa.ai cloud service."

File: README.md [hunk 3]
```diff
@@ -54,6 +54,31 @@ USERNAME=admin PASSWORD=admin ENDPOINT=https://localhost:9200 poetry run src/ope
...
+
+```sh
+docker info | grep "Total Memory" # make sure it's at least 4Gb
```
Comment (dblock) [inline README.md:63]: And if not? 

Would make it its own sentence: "Make sure you [configure Docker with at least 4GB RAM](link to source how to configure)." Check with `docker info | grep "Total Memory"`.

File: src/vespa/hello.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        {
+            "id": "vec2",
+            "values": [0.2, 0.3, 0.4],
+            "metadata": {"genre": "comedy"},
+        },
+    ]
+
+    for vector in vectors:
+        data = json.dumps({"fields": vector})
+        client.post(urljoin(endpoint, "/document/v1/vector/vector/docid/" + vector["id"]), headers=headers, data=data)
```
Comment (dblock) [inline src/vespa/hello.py:49]: Does Vespa not have a bulk interface? Just curious.

File: src/vespa/hello.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+    # search
+    query = "yql=select * from sources * where {targetHits: 1}nearestNeighbor(values,vector_query_embedding)" \
```
Comment (dblock) [inline src/vespa/hello.py]: Can we add a space before `nearestNeighbor`? Not sure it works.

File: src/vespa/hello.py [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+        "&input.query(vector_query_embedding)=[0.1,0.2,0.3]"
+
+    results = client.get(
+        urljoin(endpoint, "/search/"), headers=headers, params=query
+    ).json()
+    print(results["root"]["children"][0]["fields"])
+
+    # delete docs
+    for vector in vectors:
+        client.delete(urljoin(endpoint, "/document/v1/vector/vector/docid/" + vector["id"]), headers=headers)
```
Comment (dblock) [inline src/vespa/hello.py]: Is there a way to drop the index?

File: src/vespa/vector-app/services.xml [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+    <content id="vector" version="1.0">
+        <redundancy>2</redundancy>
+        <documents>
+            <document type="vector" mode="index" />
+        </documents>
+        <nodes>
+            <node hostalias="node1" distribution-key="0" />
+        </nodes>
+    </content>
+
```
Comment (dblock) [inline src/vespa/vector-app/services.xml]: Remove empty lines.

Also I think redundancy can be 1 and be faster, but I could be wrong ;)

**Review by dblock — CHANGES_REQUESTED** [review]
Totally awesome. The only "must have" is the TOC. Everything else is nice to have, feel free to do/not do as you wish.

File: src/vespa/hello.py (hunk 4 above)
Comment (davidlago) [reply src/vespa/hello.py:49]: For what I found, they have a [Feed CLI](https://docs.vespa.ai/en/vespa-cli-feed.html) that wraps around the [Document API](https://docs.vespa.ai/en/document-v1-api-guide.html) (the one I'm using here). I could not find a way to pass multiple documents to index directly via API calls. Perhaps looking at how the Feed CLI does it, it is using undocumented endpoints vs just iterating through the docs and calling the Document API multiple times...

File: README.md (hunk 1 above)
Comment (davidlago) [reply README.md:58]: Addressed in the commit I'm working on next, pushing shortly.

File: README.md (hunk 2 above)
Comment (davidlago) [reply README.md]: Addressed.

File: README.md (hunk 3 above)
Comment (davidlago) [reply README.md:63]: Addressed.

File: src/vespa/hello.py (hunk 5 above)
Comment (davidlago) [reply src/vespa/hello.py]: Tested, worked just fine. Updating.

File: src/vespa/hello.py (hunk 6 above)
Comment (davidlago) [reply src/vespa/hello.py]: The "index" was created as part of the application. Let me update to drop the whole application.

File: src/vespa/vector-app/services.xml (hunk 7 above)
Comment (davidlago) [reply src/vespa/vector-app/services.xml]: Removed empty lines (except the last one)

**Pull request merged**